rust/lancedb/src/index/vector.rs

```diff
@@ -46,17 +47,19 @@ impl VectorIndex {
     }
 }
 
+#[skip_serializing_none]
 #[derive(Debug, Deserialize)]
 pub struct VectorIndexMetadata {
-    pub metric_type: String,
-    pub index_type: String,
+    pub metric_type: Option<String>,
+    pub index_type: Option<String>,
 }
 
+#[skip_serializing_none]
 #[derive(Debug, Deserialize)]
 pub struct VectorIndexStatistics {
     pub num_indexed_rows: usize,
     pub num_unindexed_rows: usize,
-    pub index_type: String,
+    pub index_type: Option<String>,
```

Comment:
Can this be an enum?

Reply:
I would like to have it as enum. But it might require changes from how we write index stats for lance dataset. Currently, for a vector index, this index type is "IVF" but for a scalar index, this index type is None - ideally, it should be "BTree"